### PR TITLE
Add ability to parse "experimental" bundles in MoM

### PR DESCRIPTION
**Pull request by castulo**
Mixer will soon start using a new modifier for bundles in a MoM that are marked
as experimental. These bundles will include a "e" in the 4th position of the
modifiers of a manifest, like shown here:

`M..e 4b91c3122e7e32f1e3edb0a705a6168ca8e66f597c6f89ff32cfffd977afb5e5 10 test-bundle`

This PR adds the mechanism to parse that element from the MoM and store it in
memory. In the future this will allow swupd to treat this experimental bundles
differently if required.

### CODE REVIEW

**Comment by castulo** [comment]
This PR is part of the implementation of #676

File: src/manifest.c [hunk 1]
```diff
@@ -299,6 +299,8 @@ static struct manifest *manifest_from_file(int version, char *component, bool he
 		} else if (c[3] == 'm') {
 			file->is_mix = 1;
 			manifest->is_mix = 1;
+		} else if (c[3] == 'e') {
```
Comment (otaviobp) [inline src/manifest.c]: is this already defined with mixer-tools team? If so, I think we're good to merge this

Comment (castulo) [reply src/manifest.c]: > is this already defined with mixer-tools team? If so, I think we're good to merge this

@tmarcu @reaganlo can you guys confirm this? ^

Comment (reaganlo) [reply src/manifest.c]: @castulo Not yet. Will keep you posted when done.

Comment (matthewrsj) [reply src/manifest.c]: It is fine that it is not implemented yet, this change can still be pushed in to swupd-client as long as the mixer team agrees it is the correct approach.

Comment (matthewrsj) [reply src/manifest.c]: This approach has a +1 from me fwiw.

Comment (castulo) [reply src/manifest.c]: @reaganlo it does not need to be implemented yet, I was only asking if that is the way it is going to work. We can merge this without the implementation on the mixer side, it should not affect swupd client at all.

Comment (reaganlo) [reply src/manifest.c]: @castulo aaa I see. It looks fine. But can you please hold on to merging it until tomorrow. I want to first test and ensure that it does not break anything. Currently working on some high priority Coverity tasks.

Comment (matthewrsj) [reply src/manifest.c]: It will not break anything mixer side. This is just fixing the client to not fail if it encounters the flag.

**Comment by castulo** [comment]
ups, actually I forgot to add the test to the Makefile  :sweat_smile:

File: src/manifest.c (hunk 1 above)
Comment (reaganlo) [reply src/manifest.c]: @matthewrsj I was not referring to testing this swupd-client change. What I meant was testing the corresponding implementation in mixer.

**Comment by castulo** [comment]
There it is

File: src/manifest.c (hunk 1 above)
Comment (matthewrsj) [reply src/manifest.c]: I don't think we need to wait. Adding flags to manifests is not complicated. It is something we've done before many times.

**Pull request merged**